### PR TITLE
refactor(warehouse): utility for staging table name and some refactoring

### PR DESCRIPTION
**Pull request by achettyiitr**
# Description

Since staging table logic is distributed across packages. Instead of code repeating for constructing the staging table name use a single utility to get it.
Some code cleanup.
Some utility test cases.

## Notion Ticket

https://www.notion.so/rudderstacks/Refactor-staging-table-name-03572926ad034438b70342b889fea70f

## Security

- [x] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: warehouse/redshift/redshift.go [hunk 1]
```diff
@@ -534,9 +523,19 @@ func Connect(cred RedshiftCredentialsT) (*sql.DB, error) {
...
+		  table_name
+		from
+		  information_schema.tables
+		where
+		  table_schema = '%s'
+		  AND table_name like '%s%s';
+	`,
+		rs.Namespace,
+		warehouseutils.StagingTablePrefix(provider),
+		"%",
```
Comment (lvrach) [inline warehouse/redshift/redshift.go:537]: You can use `%%` if you want a literal `%` https://pkg.go.dev/fmt

I find this a bit more readable, although I would prefer using prepared statements here.
```suggestion
		  AND table_name like '%s%%';
	`,
		rs.Namespace,
		warehouseutils.StagingTablePrefix(provider),
```

File: warehouse/utils/utils.go [hunk 2]
```diff
@@ -139,11 +170,22 @@ var DiscardsSchema = map[string]string{
...
 
+var tableNameLimitMap = map[string]int{
+	AZURE_SYNAPSE: 127,
+	BQ:            127,
+	DELTALAKE:     127,
+	MSSQL:         127,
+	POSTGRES:      63,
+	RS:            127,
+	SNOWFLAKE:     127,
+}
```
Comment (lvrach) [inline warehouse/utils/utils.go:181]: Ideally we want those values encapsulated on their packages

Comment (achettyiitr) [reply warehouse/utils/utils.go:181]: Good point.

File: warehouse/redshift/redshift.go (hunk 1 above)
Comment (achettyiitr) [reply warehouse/redshift/redshift.go:537]: Will make the changes.

Comment (achettyiitr) [reply warehouse/redshift/redshift.go]: Done.

File: warehouse/utils/utils.go (hunk 2 above)
Comment (achettyiitr) [reply warehouse/utils/utils.go]: Done.

**Review by Jayachand — APPROVED** [review]

**Review by lvrach — APPROVED** [review]

**Pull request merged**